network/src/transport/mod.rs

```diff
@@ -96,6 +96,12 @@ impl ConnectionIdGenerator {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub enum TrustLevel {
+    Trusted,
+    Untrusted,
+}
```

Comment:
Add a small comment explaining TrustLevel::Trusted ==> peer is in trusted_peers set

Reply:
I suspect TrustLevel may change to Role, but I haven't had a chance to discuss the concept in great depth with folks. In the ideal case, we would actually distinguish a random downstream peer from a priority upstream (a private VFN), upstream (a public VFN or validator), privileged downstream (VASP, DD) and then funneling that into the connection data so that applications can learn this information.